Image uploader component

engolder opened the pull request.

engolder committed:
Add deletable image preview component

engolder committed:
Add image uploader component

engolder committed:
Fix wrong method

image uploader can't receive file if it have file through prop already.
but it was not.

khg0712 commented on src/presentations/ImageUploader/index.tsx:
```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import * as S from './styles';
+import { DeletableImagePreview, Button } from '..';
```
'..' 이렇게 경로 써넣으니까 뭔가 똥싸다가 끊긴 느낌인데 어떻게 생각하심?

engolder replied:
음..?

khg0712 commented on src/presentations/ImageUploader/index.tsx:
```diff
@@ -0,0 +1,104 @@
+}
+
+interface ImageUploaderState {
+  onDraggingOver: boolean;
```
onChange, onDelete는 모두 진행형이 아닌데 이곳만 진행형으로 써둔 이유가 있나요?

engolder replied:
web drag api 명칭입니다. Drag와 Dragging은 서로 다른 이벤트입니다

engolder replied:
onDraggingOver는 이벤트가 아닙니다. boolean prop입니다

khg0712 commented on src/presentations/ImageUploader/index.tsx:
```diff
@@ -0,0 +1,104 @@
+    super(props);
+
+    this.state = {
+      onDraggingOver: false,
```
진행형!

khg0712 commented on src/presentations/ImageUploader/index.tsx:
```diff
@@ -0,0 +1,104 @@
+
+  handleDragLeave = (event: React.DragEvent<HTMLLabelElement>) => {
+    prevent(event);
+
```
무의미한 enter!

khg0712 commented on src/presentations/ImageUploader/index.tsx:
```diff
@@ -0,0 +1,104 @@
+
+  render() {
+    const { onDelete, id, image, isCoverImage = false } = this.props;
+    const { onDraggingOver } = this.state;
```
진행형!

khg0712 commented on src/presentations/ImageUploader/styles.ts:
```diff
@@ -0,0 +1,57 @@
+  isActive: boolean;
+}
+
+export const TableCell = styled.div`
```
가운데 정렬할 때 flex를 쓰면 좋아용! 그리구 TableCell이라는 네이밍이 조금 그런 것 같아용 더 나은 네이밍을 찾아봅시다!

engolder committed:
reformat wrap

engolder committed:
Fix styling method

I used flex attr to make vertical center instead table display attr.

flex is better than table and table cell for readablity

engolder committed:
Merge branch 'master' of https://github.com/daemabnb/Daema-BnB-web into image-uploader-component
